src/v/cluster/node_status_backend.cc

```diff
@@ -273,6 +265,23 @@ node_status_backend::process_reply(result<node_status_reply> reply) {
           err.message());
         return err;
     }
+
+    _stats.rpcs_sent += 1;
+    auto& replier_metadata = reply.value().replier_metadata;
+    if (replier_metadata.node_id != target_node_id) {
```

Comment:
why do we only check it on reply, but not on request received in `node_status_backend::process_request` or in `node_status_rpc_handler::node_status`, like we do it in `service::do_collect_node_health_report`?

Reply:
i think in this case it is not that required as replying the status request is very cheap